mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp

```diff
@@ -14,34 +15,100 @@
  */
 --%>
 
-<%@ include file="/init.jsp" %>
+<%@ include file="init.jsp" %>
 
 <div class="list-container">
 	<div class="header">
 		<h2>
 			<liferay-ui:message key="to-do-list" />
-
 			<i class="icon-check"></i>
 		</h2>
 	</div>
+	<div class="welcome-message">
+		<liferay-ui:message key="welcome" />
+		<%= user.getFirstName() %>
+		<img alt="<%= user.getFullName() %>" src="<%=user.getPortraitURL(themeDisplay)%>">
+	</div>
+	<liferay-ui:tabs names="tab-1,tab-2,tab-3" refresh="false" >
 
-	<div class="task-container">
-		<aui:input cssClass="input-item" name="task" placeholder="Enter your task.." />
+		<liferay-ui:section>
+		<div class="task-container">
+			<aui:form>
+				<aui:input cssClass="input-item" name="task" placeholder="enter-task">
+					<aui:validator name="required" />
+				</aui:input>
 
-		<aui:button cssClass="add-task" name="add" value="add" />
+				<aui:button cssClass="add-task" name="add" value="add" />
+			</aui:form>
 
-		<span class="counter">35</span>
+			<span class="counter">35</span>
 
-		<liferay-ui:message key="characters-remaining" />
+			<liferay-ui:message key="characters-remaining" />
+			<h3>
+				<liferay-ui:message key="tasks" />
+			</h3>
 
-		<ul class="task-list">
-			<li>
-				<liferay-ui:message key="sample-task" />
+			<ul class="task-list">
+				<li>
+					<liferay-ui:message key="sample-task" />
 
-				<aui:button cssClass="delete-task" icon="icon-remove" size="large" />
-			</li>
-		</ul>
-	</div>
+					<aui:button cssClass="delete-task" icon="icon-remove" size="large" />
+				</li>
+			</ul>
+			<div class="remaining-tasks-container">
+				<div class="remaining-tasks-count">
```

Comment:
You have many many empty elements. This practice should be an exception, used only is special circumstances like you need a container for ajax response or it is a standalone icon ```<i>``` tag.